src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonArrayFormatter.java

```diff
@@ -0,0 +1,358 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jp.co.yahoo.yosegi.message.formatter.json;
+
+import jp.co.yahoo.yosegi.message.parser.IParser;
+import jp.co.yahoo.yosegi.message.parser.json.JacksonNullParser;
+import jp.co.yahoo.yosegi.message.objects.PrimitiveObject;
+import jp.co.yahoo.yosegi.message.objects.StringObj;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.Arguments;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TestJacksonArrayFormatter {
+
+  private class TestParentParser implements IParser {
+
+    private final boolean isArray;
+
+    public TestParentParser( final boolean isArray ) {
+      this.isArray = isArray;
+    }
+
+    @Override
+    public PrimitiveObject get(final String key ) throws IOException {
+      return null;
+    }
+
+    @Override
+    public PrimitiveObject get( final int index ) throws IOException {
+      return null;
+    }
+
+    @Override
+    public IParser getParser( final String key ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public IParser getParser( final int index ) throws IOException {
+      if ( index == 0 ) {
+        return new TestPrimitiveParser( isArray );
+      }
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public String[] getAllKey() throws IOException {
+      return new String[]{};
+    }
+
+    @Override
+    public boolean containsKey( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public int size() throws IOException {
+      return 1;
```

Comment:
Why in the case of containsKey returns false, size() returns one?
It seems strange for me.

Reply:
Because this parser only holds index 0, its size is 1.

Reply:
Thank you.